scripts/scil_tractogram_filter_collisions.py

```diff
@@ -0,0 +1,174 @@
+import numpy as np
+
+from scilpy.tractograms.intersection_finder import IntersectionFinder
+from dipy.io.stateful_tractogram import StatefulTractogram
+from dipy.io.streamline import save_tractogram
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (assert_inputs_exist,
+                             assert_outputs_exist,
+                             add_overwrite_arg,
+                             add_verbose_arg,
+                             add_reference_arg,
+                             add_bbox_arg)
+from scilpy.tracking.fibertube import add_random_options
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description=__doc__)
+
+    p.add_argument('in_tractogram',
+                   help='Path to the tractogram file containing the \n'
+                   'streamlines (must be .trk or .tck).')
+
+    p.add_argument('in_diameters',
+                   help='Path to a text file containing a list of \n'
+                   'diameters in mm. Each line corresponds \n'
+                   'to the identically numbered streamline.')
+
+    p.add_argument('out_tractogram',
+                   help='Tractogram output file (must be .trk or \n'
+                   '.tck). Another file (.txt) containing the diameters \n'
+                   'will be created using the same file name with \n'
+                   '"_diameters" appended.')
+
+    p.add_argument('--single_diameter', action='store_true',
+                   help='If set, the first diameter found in \n'
+                   '[in_diameters] will be repeated for each fiber.')
+
+    p.add_argument('-cr', '--save_colliders', action='store_true',
+                   help='If set, the script will produce another \n'
+                   'tractogram (.trk) containing only streamlines that \n'
+                   'have filtered out (with their collision point). \n'
+                   'Its file name is derived from the out_tractogram \n'
+                   'parameter with "_colliders" appended.')
+
+    p.add_argument('-cd', '--save_collided', action='store_true',
+                   help='If set, the script will produce another \n'
+                   'tractogram (.trk) containing only valid streamlines \n'
+                   'that have been collided by a filtered one. Its file \n'
+                   'name is derived from the out_tractogram parameter with \n'
+                   '"_collided" appended.')
+
+    metrics_g = p.add_argument_group('Metrics options')
+    metrics_g.add_argument('--tmv_size_threshold', default=None, type=float,
+                           help='If set, fibers that don\'t collide \n'
+                           'will be filtered further, so that the edge \n'
+                           'length of true_max_voxel is below the given \n'
+                           'threshold. (see ft_fibers_metrics.py)'
+                           '[%(default)s]')
```

Comment:
I have no idea what this parameter is and what it does. 
- what does TMV mean? Define the acronym
- what is true_max_voxel ?
- Do you really need this parameter? If so, you need to better explain it.

Reply:
I changed it completely and now it's a lot more clear.